iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -140,15 +143,23 @@ public static Table createTable(Configuration conf, Properties props) {
     Map<String, String> map = filterIcebergTableProperties(props);
 
     Optional<Catalog> catalog = loadCatalog(conf, catalogName);
-
+    SortOrder sortOrder = getSortOrder(props, schema);
     if (catalog.isPresent()) {
       String name = props.getProperty(NAME);
       Preconditions.checkNotNull(name, "Table identifier not set");
-      return catalog.get().createTable(TableIdentifier.parse(name), schema, spec, location, map);
+      return catalog.get().buildTable(TableIdentifier.parse(name), schema).withPartitionSpec(spec)
+          .withLocation(location).withProperties(map).withSortOrder(sortOrder).create();
     }
 
     Preconditions.checkNotNull(location, "Table location not set");
-    return new HadoopTables(conf).create(schema, spec, map, location);
+    return new HadoopTables(conf).create(schema, spec, sortOrder, map, location);
+  }
+
+  private static SortOrder getSortOrder(Properties props, Schema schema) {
+    String sortOrderJsonString = props.getProperty(InputFormatConfig.INSERT_WRITE_ORDER);
```

Comment:
````
return SortOrderParser.fromJson(schema(), sortOrderAsJson)
````

Reply:
I think I would use  "asJson"  in a method name to express a transformation is happening like toString(), asJson(), asXML ect not in a variable name.

Reply:
i don't insist, just saw this naming is used in iceberg